intellij-plugin-structure/structure-classes/src/main/java/com/jetbrains/plugin/structure/classes/utils/KtClassResolver.kt

```diff
@@ -0,0 +1,71 @@
+package com.jetbrains.plugin.structure.classes.utils
+
+import kotlinx.metadata.jvm.KotlinClassMetadata
+import kotlinx.metadata.jvm.Metadata
+import org.objectweb.asm.tree.AnnotationNode
+import org.objectweb.asm.tree.ClassNode
+import java.util.*
+
+internal const val KOTLIN_METADATA_ANNOTATION_DESC = "Lkotlin/Metadata;"
+
+class KtClassResolver {
+  private val cache = TreeMap<ClassNode, KtClassNode>(Comparator<ClassNode> { o1, o2 ->
+    val s1 = o1?.signature ?: ""
+    val s2 = o2?.signature ?: ""
+    s1.compareTo(s2)
+  })
+
+  operator fun get(classNode: ClassNode): KtClassNode? {
+    return cache.getOrPut(classNode) {
+      return findMetadataAnnotation(classNode)?.let {
```

Comment:
Please note that unqualified `return` statement in lambda expressions passed to inline function returns from the outer functions (see [the doc](https://kotlinlang.org/docs/returns.html#return-to-labels)). So actually this code never puts data to the cache. In order to return value from the lambda, you may either use qualified return or simply remove `return` statement (the last expression in the lambda is returned automatically).

Reply:
Well spotted, thanks for insights!